e2e/LongHaul/device/IotHub.cs

```
@@ -120,16 +122,7 @@ public async Task SendTelemetryMessagesAsync(Logger logger, CancellationToken ct
                 if (!IsConnected
                     || !_messagesToSend.Any())
                 {
-                    try
```

We can let it throw to the top level like we do elsewhere.